nixos/modules/system/boot/systemd-lib.nix

```diff
@@ -78,6 +78,10 @@ in rec {
     optional (badFields != [ ])
       "Systemd ${group} has extra fields [${concatStringsSep " " badFields}].";
 
+  assertInt = name: group: attr:
+    optional (attr ? ${name} && !isInt attr.${name})
+      "Systemd ${group} field `${name}' is not an integer";
+
```

Comment:
`isInt` allows any integer, even negative, so this significantly weakens the assertion too much.
You should keep checking for a range with `assertRange` but replace the hardcoded upper bound `4294967295` with the largest positive integer depending on the platform's bitsize.

Reply:
Why not just checking if it is larger then 0?

Reply:
In this particular case there may be a simpler solution: just check for `isInt` and the lower bound , so you don't have to specify the upper bound.

Reply:
(guess we were writing in parallel :smile:)